src/components/screens/form-edit-screen.tsx

```diff
@@ -327,95 +467,345 @@ export class FormEditScreen extends React.Component<Props, State> {
-        { listOfFields.map((field: string, index: number) => (
-          <Box border={ 1 } className={ classes.fields } key={ index } >
-            <ListItemText>
-              <Typography>{ field }</Typography>
-            </ListItemText>
-          </Box>
-        )) }
-      </List>
+      <Box className={ classes.componentContainer }>
+        <Droppable droppableId="componentList" isDropDisabled>
+          {(provided:DroppableProvided, snapshot:DroppableStateSnapshot) => (
+            <div
+              ref={ provided.innerRef }
+            >
+              { Object.values(MetaformFieldType).map(this.renderDraggableComponent) }
+            </div>
+          )}
+        </Droppable>
+      </Box>
     );
   }
 
   /**
-   * Method for rendering addable components
+   * Renders a single draggable component
+   * 
+   * @param fieldType field type
+   * @param index index
    */
-  private renderComponents = () => {
-    const listOfComponents = [
-      strings.formEditScreen.dropDownMenu,
-      strings.formEditScreen.selectBox,
-      strings.formEditScreen.radioButton,
-      strings.formEditScreen.button,
-      strings.formEditScreen.image
-    ];
-
+  private renderDraggableComponent = (fieldType: MetaformFieldType, index: number) => {
     const { classes } = this.props;
 
     return (
-      <List>
-        <Box border={ 1 } className={ classes.componentHeader }>
-          <ListItemText>
-            <Typography>
-              { strings.formEditScreen.leftSideBarComponentHeader }
-            </Typography>
-          </ListItemText>
-        </Box>
-        { listOfComponents.map((component: string, index: number) => (
-          <Box border={ 1 } className={ classes.fields } key={ index }>
-            <ListItemText>
-              <Typography>{ component }</Typography>
-            </ListItemText>
-          </Box>
-        )) }
-      </List>
+      <Draggable draggableId={ fieldType } index={ index }>
+        {(providedDraggable:DraggableProvided, snapshotDraggable:DraggableStateSnapshot) => (
+            <>
+              <div
+                className={ classes.singleDraggableComponent }
+                ref={ providedDraggable.innerRef }
+                { ...providedDraggable.draggableProps }
+                { ...providedDraggable.dragHandleProps }
+              >
+                { this.renderReadOnlyInput(fieldType) }
+              </div>
+              { snapshotDraggable.isDragging &&
+                <div 
+                  className={ classNames(classes.singleDraggableComponent, { clone : true }) }
+                > 
+                  { this.renderReadOnlyInput(fieldType) }
+                </div> 
+              }
+            </>
+          )}
+      </Draggable>
     );
   }
 
+  /**
+   * Event handler for drag start
+   * 
+   * @param initial drag start data
+   * @param provided responder provided
+   */
+  private onDragStart = (initial: DragStart, provided: ResponderProvided) => {
```

Comment:
Form edit screen component is getting pretty big. Maybe should consider splitting the code. DnD methods could be located on separate file for example.